src/proxy.c

```diff
@@ -238,113 +239,131 @@ cc_proxy_msg_read (GIOChannel *source, GIOCondition condition,
 	struct proxy_msg p_msg = { 0 };
 	gchar iov_buffer[LINE_MAX] = { 0 };
 	ssize_t bytes_read;
-	gboolean fd_read = false;
 	struct msghdr msg = { 0 };
 	struct cmsghdr *cmsg = NULL;
 	struct iovec io = { .iov_base = iov_buffer,
 	                    .iov_len = sizeof(iov_buffer) };
-	char ctl_buffer[LINE_MAX] = { 0 };
+	char ctl_buffer[CMSG_SPACE(sizeof(int))] = { 0 };
 	unsigned char *data;
 	/* length + reserved */
 	const size_t data_offset = sizeof(p_msg.length) + sizeof(p_msg.reserved);
 	guint8 buf_len[sizeof(p_msg.length)] = { 0 };
-	size_t bytes_to_copy = 0;
+	unsigned char *frame = NULL;
+	unsigned char *tmp_ptr;
+	int timeout;
+	size_t tot_b_rd = 0;
+	size_t tot_b_expected;
+	ssize_t b_to_cpy;
 
 	msg.msg_iov = &io;
 	msg.msg_iovlen = 1;
 	msg.msg_control = ctl_buffer;
 	msg.msg_controllen = sizeof(ctl_buffer);
 
-	/* read length */
-	bytes_read = recvmsg(proxy_data->socket_fd, &msg, 0);
-
-	if (bytes_read == -1){
-		g_critical("failed to read proxy message %s", strerror(errno));
-		return false;
+	/* 
+	 * This sleep is very important to make sure we don't
+	 * get MSG_CTRUNC message from msghdr. There is no way
+	 * to recover from a MSG_CTRUNC the control message lost.
+	 */
+	if (proxy_data->oob_fd) {
+		usleep(1e3);
```

Comment:
Well, if you read my commit message and the comment on top of that usleep(), it is because of MSG_CTRUNC is received. If you have a good understanding about why I receive this MSG_CTRUNC sometimes without the usleep(), let me know, because I really don't get what is the reason...
I 100% agree it is fragile, and I would expect a better solution if I was able to understand the real root cause.
About the value, I tried with 1s, 100ms, 1ms, and it was still working. I didn't have time to try 100us, 10us and 1us yet. But the point is that without understanding the root cause, I don't know if it will work on all systems.
I know on mine it is solving this issue and issue #522 too.